File: iterative.go
package main

import "fmt"

func factorial(n int) int {
  var a int = 1
  for i := 1; i <= n; i++ {
    a = a * i
  }
  return a
}

func main() {
  var input int
  fmt.Print("Number: ")
  fmt.Scanln(&input)
  fmt.Println(factorial(input))
}

File: iterative.kt
import java.util.Scanner

fun factorial(n: Int): Int {
    var a: Int = 1
    for(i in 1..n) {
        a = a * i
    }
    return a
}

fun main() {
    val reader = Scanner(System.`in`)
    print("Number:")
    var input: Int = reader.nextInt()
    var result: Int = factorial(input)
    println("$result")
}
